src/python/pants/jvm/test/junit.py

```diff
@@ -214,4 +221,5 @@ def rules():
         *lockfile.rules(),
         UnionRule(TestFieldSet, JunitTestFieldSet),
         UnionRule(GenerateToolLockfileSentinel, JunitToolLockfileSentinel),
+        *JunitTestRequest.registration_rules(),
```

Comment:
Commenting out this line gets at least 1 of the failing Java-related tests to pass. I can't figure out what's different between the JVM rule structure / test setup and everything else that is seemingly working fine 🤯